hangman.py

```diff
@@ -0,0 +1,250 @@
+
+
+wordlist = load_words()
+word = choose_word(wordlist)
+
+
+def get_total_score(secret_word, guesses_remaining):
+    """
+    secret_word: string, the word the user is guessing; assumes all letters are
+      lowercase
+    guesses_remaining: guesses that are left
+    returns: user's score;
+
+    """
+
+    total_score = len(set(secret_word)) * guesses_remaining
+    return total_score
+
+
+def is_word_guessed(secret_word, letters_guessed):
+    """
+    secret_word: string, the word the user is guessing; assumes all letters are
+      lowercase
+    letters_guessed: list (of letters), which letters have been guessed so far;
+      assumes that all letters are lowercase
+    returns: boolean, True if all the letters of secret_word are in letters_guessed;
+      False otherwise
+    """
+
+    if set(secret_word) == set(letters_guessed):
+        print('-' * 20, f'\nCongratulations, you won!\nTotal score: {get_total_score(word, guesses)}')
+        return True
+    else:
+        return False
+
+
+def get_guessed_word(secret_word, letters_guessed):
+    """
+    secret_word: string, the word the user is guessing
+    letters_guessed: list (of letters), which letters have been guessed so far
+    returns: string, comprised of letters, underscores (_), and spaces that represents
+      which letters in secret_word have been guessed so far.
+    """
+    result = []
+    for char in secret_word:
+        if char not in letters_guessed:
+            result.append(UNKNOWN_LETTER)  # append a "_" symbol to a word
+        elif char in letters_guessed:
+            result.append(char)  # append guessed letters to a word
+    return ''.join(result)
+
+
+def get_available_letters(letters_guessed):
+    """
+    letters_guessed: list (of letters), which letters have been guessed so far
+    returns: string (of letters), comprised of letters that represents which letters have not
+      yet been guessed.
+    """
+
+    result = []
+    for i in letters:
+        if i not in letters_guessed:
+            result.append(i)
+    return ''.join(result)
+
+
+def is_ascii(user_guess):
+    """
+    user_guess: input the user is trying to guess;
+    returns: True if input is letter, otherwise False
+    """
+    is_letter = user_guess != '' and user_guess.lower() in ascii_lowercase  # checks if input in ascii code
+    return is_letter
+
+
+def get_warnings(user_guess, my_word):
+    """
+    Func checks user input for invalid values and repeated letters;
+    user_guess: input the user is trying to guess;
+    my_word: string with _ characters, current guess of secret word;
+    returns: boolean - True and append letter to list available letters,
+    otherwise False, print information message about
+    repeated  letter or use of a banned symbol
+    and take away warnings;
+    """
+    global letters, letters_guessed, warnings, guesses
+
+    is_warning = is_ascii(user_guess) and user_guess in letters  # checks if input a valid letter
```

Comment:
Effectively all letters in `letters` are ascii, so there's no need in first part of condition. But a check `user_guess in letters` is slower. It's better to use methods `is_ascii` and `is_alpha`